### PR TITLE
Add NCES ID support per APHL guidance

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
This PR implements the APHL guidance for schools as the ordering facility. The APHL recommendation uses the National Center for Educational Statistics (NCES) identifier. 

## Changes
- NCES ID table added school name and zip code as index
- `useNCESFacilityName` flag in HL7 translation setting
- `lookupBestMatch` method that handles non-exact matches
- How to doc on how to get the NCES table information
- Release note entry
- OpenAPI.YML entry

Reviewers should look at the `lookupBestMatch` algorithm. 

### Testing
- [x] Tested locally?
- [x] Ran `quickTest all`?
- [x] Ran `./prime test` against local Docker ReportStream container?
- [ ] Downloaded a file from `http://localhost:7071/api/download`?
- [x] Added tests?

### Specific Security-related subjects a reviewer should pay specific attention to

- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [x] Includes a summary of what a code reviewer should verify?
- [x] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Fixes
*List GitHub issues this PR fixes*
- #2105

### CODE REVIEW

File: prime-router/src/main/kotlin/LookupTable.kt [hunk 1]
```diff
@@ -293,11 +381,15 @@ class LookupTable(
...
     companion object {
         fun read(fileName: String): LookupTable {
-            return read(File(fileName).inputStream())
+            val file = File(fileName)
+            return read(file.inputStream(), isTsv = file.extension == "tsv")
         }
 
-        fun read(inputStream: InputStream): LookupTable {
-            val table = csvReader().readAll(inputStream)
+        fun read(inputStream: InputStream, isTsv: Boolean = false): LookupTable {
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/LookupTable.kt:388]: Adding support for TSV files. At one time, the NCES file was a TSV file.

File: prime-router/src/main/kotlin/serializers/Hl7Serializer.kt [hunk 2]
```diff
@@ -533,6 +536,9 @@ class Hl7Serializer(
                         setAOE(terser, element, aoeSequence++, date, "UNK", report, row, suppressQst = suppressQst)
                     }
                 }
+            } else if (element.hl7Field == "ORC-21-1") {
```
Comment (RickHawesUSDS) [inline prime-router/src/main/kotlin/serializers/Hl7Serializer.kt:539]: Following the pattern for special-casing a particular HL7 field

File: prime-router/docs/openapi.yml [hunk 3]
```diff
@@ -837,6 +837,10 @@ components:
           description: The name of the receiving organization in the message
           type: string
           default: null
+        useNCESFacilityName:
+          description: Lookup NCES ID and follow APHL guidance for ORC-21
+          type: boolean
+          default: false
         transport:
```
Comment (RickHawesUSDS) [inline prime-router/docs/openapi.yml:844]: Document the new field

File: prime-router/src/test/unit_test_files/pdi-covid-19-wa-k12.csv [hunk 4]
```diff
@@ -0,0 +1,15 @@
...Instrument_ID,Result_ID,Corrected_result_ID,Test_result_status,Test_result_code,Illness_onset_date,Specimen_collection_date_time,Order_test_date,Test_date,Date_result_released,Patient_race,Patient_DOB,Patient_gender,Patient_ethnicity,Patient_preferred_language,Patient_street,Patient_street_2,Patient_city,Patient_state,Patient_zip_code,Patient_phone_number,Patient_county,Patient_email,Patient_role,Processing_mode_code,Employed_in_healthcare,Resident_congregate_setting,First_test,Symptomatic_for_disease,Testing_lab_name,Testing_lab_CLIA,Testing_lab_street,Testing_lab_street_2,Testing_lab_city,Testing_lab_state,Testing_lab_zip_code,Testing_lab_phone_number,Testing_lab_county,Organization_name,Ordering_facility_name,Ordering_facility_street,Ordering_facility_street_2,Ordering_facility_city,Ordering_facility_state,Ordering_facility_zip_code,Ordering_facility_phone_number,Ordering_facility_county,Ordering_facility_email,Ordering_provider_ID,Ordering_provider_last_name,Ordering_provider_first_name,Ordering_provider_street,Ordering_provider_street_2,Ordering_provider_city,Ordering_provider_state,Ordering_provider_zip_code,Ordering_provider_phone_number,Ordering_provider_county,Site_of_care
```
Comment (RickHawesUSDS) [inline prime-router/src/test/unit_test_files/pdi-covid-19-wa-k12.csv:1]: This test data has been deindentified.

**Review by MauriceReeves-usds — APPROVED** [review]

**Pull request merged**